lnwallet/channel.go

```diff
@@ -5591,19 +5584,12 @@ func extractHtlcResolutions(feePerKw SatPerKWeight, ourCommit bool,
 		// If the HTLC is incoming, then we'll attempt to see if we
 		// know the pre-image to the HTLC.
 		if htlc.Incoming {
-			// We'll now query the preimage cache for the preimage
-			// for this HTLC. If it's present then we can fully
-			// populate this resolution.
-			preimage, _ := pCache.LookupPreimage(htlc.RHash)
-
-			// Otherwise, we'll create an incoming HTLC resolution
-			// as we can satisfy the contract.
-			var pre [32]byte
-			copy(pre[:], preimage[:])
+			// We'll create an incoming HTLC resolution which allows
+			// us to satisfy the contract assuming the preimage is
+			// discovered before the timelock expires.
```

Comment:
Am I wrong in thinking that the write consistency fix is unrelated to this change?

I'm a bit uneasy doing critical changes like this in a big PR. I would prefer having this in a separate PR where the proper discussion can be had.